functions/src/requestCreated/index.js

```diff
@@ -58,48 +74,135 @@ async function sendFcms(userUids) {
+      // .limit(50)
+      .get(),
+  );
+
+  if (nearbyUsersErr) {
     console.error(
-      `Error loading settings doc: ${settingsDocErr.message || ''}`,
-      settingsDocErr,
+      `Error querying for nearby users for request "${requestId}":`,
+      nearbyUsersErr.message,
     );
+    throw nearbyUsersErr;
+  }
+
+  console.log(
+    `Nearby user results for request "${requestId}":`,
+    nearbyUsersSnaps.docs.map((nearbyUserDoc) => nearbyUserDoc.id),
+  );
+
+  // Exit if no users found nearby
+  // TODO: Discuss switching this searching again within a bigger radius
+  if (!nearbyUsersSnaps.docs.length) {
+    console.log('No nearby users found, exiting');
+    return null;
   }
 
-  // Notify all users in newRequests parameter of system_settings/notification document
-  const { newRequests: userUids } = settingsDocSnap.data() || {};
-  await sendFcms(userUids);
+  // Load user accounts of nearby users (to check notification settings)
+  // TODO: Look into if a where with ids and notification settings performs better
+  const nearbyUsersAccountSnaps = await Promise.all(
+    nearbyUsersSnaps.docs.map((userSnap) =>
+      admin.firestore().doc(`${USERS_COLLECTION}/${userSnap.id}`).get(),
+    ),
+  );
+  console.log('Nearby users accounts loaded:', nearbyUsersAccountSnaps.length);
+  console.log(
+    `Nearby user accounts for request "${requestId}":`,
+    nearbyUsersAccountSnaps.map((nearbyUserDoc) => nearbyUserDoc.data()),
+  );
+  // Filter nearby users into ids of users with emails or browser notifications enabled
+  const { browserNotifUids, emailNotifUids } = nearbyUsersAccountSnaps.reduce(
+    (acc, userDocSnap) => {
+      // Browser notifications to users which have browser notifications enabled
+      if (userDocSnap.get('browserNotifications')) {
+        acc.browserNotifUids.push(userDocSnap.id);
+      }
+      // Email notifications to users which have super-admin role
+      if (userDocSnap.get('role') === 'super-admin') {
+        acc.emailNotifUids.push(userDocSnap.id);
+      }
+      return acc;
+    },
+    {
+      browserNotifUids: [],
+      emailNotifUids: [],
+    },
+  );
+  console.log('Notification settings:', {
+    browserNotifUids,
+    emailNotifUids,
+  });
+
+  // Exit if none of the users within range have notifications enabled
+  if (!browserNotifUids.length && !emailNotifUids.length) {
+    console.log('No users have browser notifications enabled. Exiting...');
+    return null;
+  }
 
   const projectId = getFirebaseConfig('projectId');
   // Set domain as frontend url if set, otherwise fallback to Firebase Hosting URL
   const projectDomain = getEnvConfig('frontend.url', `${projectId}.web.app`);
+  // Send FCMs to all of the nearby users with browser notification setting enabled
+  await sendFcmsToUsers(browserNotifUids, {
+    message: `Request created by ${firstName} near ${generalLocationName}`,
+    click_action: `https://${projectDomain}/requests/${requestId}`,
```

Comment:
I tested this functionality and it still doesn't seem to work.  Can you please take a look?  I think we will likely need the notification click event function.

Reply:
I think that it needs to be deployed into next before we can effectively test. Did you manually deploy it?

Reply:
I did.